extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -329,7 +404,14 @@ int ConsumeWindowsEventLog::processQueue(const std::shared_ptr<core::ProcessSess
     session->putAttribute(flowFile, FlowAttributeKey(MIME_TYPE), "application/xml");
     session->getProvenanceReporter()->receive(flowFile, provenanceUri_, getUUIDStr(), "Consume windows event logs", 0);
     session->transfer(flowFile, Success);
-    session->commit();
+
+	flowFile = session->create();
+
+	session->write(flowFile, &WriteCallback(evt.rendered_text_));
+	session->putAttribute(flowFile, FlowAttributeKey(MIME_TYPE), "text/plain");
+	session->getProvenanceReporter()->receive(flowFile, provenanceUri_, getUUIDStr(), "Consume windows event logs", 0);
+	session->transfer(flowFile, Success);
```

Comment:
I think producing two different flowfiles and transferring them to the same relationship is far from ideal. 
First it requires an additional processor to separate them, but also produces two flowfiles from one event. As the user most probably needs one or the other, this results in a lot of needless IO ops (writing flowfiles to content repo) and filling the disk without any value added. 
To resolve I would either:

- create two different relationships (text and xml) and route them accordingly. Not the best as it solves only one part of the issue. 
- introduce a processor property called "mode" or sg alike, which can be set to "text", "xml" or "both". This would support the current behaviour as well, and it case the user selects this, his is free to split them using routeonattribute, but at least we don't force that. 

As this processor is not released yet, we don't break anything. I would really appreciate to put more effort in this now and avoid being limited by backward compatibility later. 
@bakaid @am-c-p-p @phrocker @apiri - please don't hesitate to share your opinion!

Reply:
I don't agree and think that we want to keep in line with the success output which is what we have in java cwel. that was a pain point for those. i don't think what you propose solves your issues.

Reply:
I will add that I think @am-c-p-p is taking the reigns on this and he can disagree, but we should be cognizant of creating confusion between java and cpp variants. this I think yields the lowest confusion with allowing downstream processors to split or take care of this. that is why roa exists.

Reply:
@phrocker : if you check the second bulletpoint above, that's about keeping success, but making the output configurable.
Don't get me wrong, I have no issue with keeping the current behavior as an option, maybe even as a default option, but I don't feel it right to force the user to use ROA in case he only needs one and not the other.
That's why I would introduce a property with 3 options:

- Both (default, current behaviour)
- Text only
- XML only

To keep aligned with NiFi, I'm happy to add this feature to the Java codebase as well.